indexer/services/comlink/src/controllers/api/v4/orders-controller.ts

```diff
@@ -114,7 +123,28 @@ class OrdersController extends Controller {
       ),
     ]);
 
-    const postgresOrderMap: PostgresOrderMap = _.keyBy(postgresOrders, OrderColumns.id);
+    const redisOrderIds: string[] = _.map(
+      Object.values(redisOrderMap),
+      (redisOrder: RedisOrder) => {
+        return OrderTable.orderIdToUuid(redisOrder.order!.orderId!);
+      },
+    );
+
+    // Fetch additional orders from Postgres that are not in the initial query, because they
+    // might not conform to the status filter above. For example, if the user is querying for
+    // `status: [BEST_EFFORT_OPENED]`, we need to fetch all orders from Postgres, because if the
+    // order in postgres is 'OPENED', then we do not want to return this order to the user as
+    // 'BEST_EFFORT_OPENED'.
+    const additionalPostgresOrders: OrderFromDatabase[] = await OrderTable.findAll({
```

Comment:
nit: This query only needs to be done if the user is querying for orders with status including `BEST_EFFORT_OPENED` right? There's no need to do this extra SQL query if the user is querying for any other status, as they will definitely exist in postgres.

Reply:
No need to filter by status. `RedisOrder`s do not contain a status field, we assume they are BEST_EFFORT_OPENED unless there is a corresponding postgres order that says otherwise

Reply:
While redis is not the source of truth for status, we do use it as source of truth for other values mentioned here:
```
    size: redisOrder.size,
    price: redisOrder.price,
    goodTilBlock: protocolTranslations.getGoodTilBlock(redisOrder.order!)?.toString() ?? undefined,
    goodTilBlockTime: protocolTranslations.getGoodTilBlockTime(redisOrder.order!) ?? undefined,
    timeInForce: apiTranslations.orderTIFToAPITIF(redisOrderTIF),
    postOnly: apiTranslations.isOrderTIFPostOnly(redisOrderTIF),
    reduceOnly: redisOrder.order!.reduceOnly,
    ```